pysonnet/projects.py

```diff
@@ -406,6 +417,115 @@ def make_sonnet_file(self, file_path, clean=True):
+        initiate the export. Allowed keywords are listed below.
+
+        :keyword frequency: float value given in Hz
+        :keyword region_style: 'Whole' or 'Rect'
+        :keyword driving_port: port number you wish to drive
+        :keyword bottom: The lower bound of the region
+        :keyword top: The upper bound of the region
+        :keyword left: The left bound of the region
+        :keyword right: The right bound of the region
+        :keyword levels_start: The beginning metal layer to export
+        :keyword levels_stop: The ending metal layer to export
+        :keyword measurement_type: 'jxy', 'jx', 'jy'
+        :keyword measurement_complex: 'yes' or 'No'
+
+        """
+
+
+        if [left,right,top,bottom] == [None,None,None,None]:
+            region_style = 'Whole'
+        else:
+            region_style = 'Rect'
+
+        # measurement type
+        if measurement_type is None:
+            type_error = "Set measurement type as 'jxy', 'jx', or 'jy'"
+            raise ValueError(type_error)
+
+        # make default one port
+        # grab the other parameters from sonnet file
+        # kwarg should be 'which port am i driving'
+
+        # Port1 parameters
+        capacitance = self.port['capacitance']
+        inductance = self.port['inductance']
+        number = self.port['number']
+        reactance = self.port['reactance']
+        resistance = self.port['resistance']
+
+
+        frequency = str(int(frequency))
+
+        if self['geometry']['x_cells2'] is None:
+            missing_setup_box = 'run setup_box() method for cell sizes'
+            raise ValueError(missing_setup_box)
+
+        # xml file generation
+        JXY_Export_Set = ET.Element("JXY_Export_Set")
+        JXY_Export = ET.SubElement(JXY_Export_Set, "JXY_Export", Filename=self.file_name +'.csv', Label=son_label)
+        ET.SubElement(JXY_Export, "Region", Style=region_style)
+        ET.SubElement(JXY_Export, "Levels", Stop=levels_stop, Range='Some', Start=levels_start)
+        ET.SubElement(JXY_Export, "Grid", XStep=self['geometry']['x_cells2'], YStep=self['geometry']['y_cells2'])
+        ET.SubElement(JXY_Export, "Measurement", Complex=measurement_complex, Type=measurement_type)
+        Drive = ET.SubElement(JXY_Export, "Drive")
+
+        # i am trying to create a loop that sets the amount of ports and also includes setting
+        # voltage=1 to the port the user wants to drive and voltage=0 too all other ports.
+        # Not sure how to make such a dynamic loop
+        for port_num in range(number):
+            ET.SubElement(Drive, "JXYPort", Capacitance=capacitance, Inductance=inductance,
+                Number=port_num,Phase=phase, Reactance=reactance, Resistance=resistance, Voltage=driving_port)
+
+        Locator = ET.SubElement(JXY_Export, "Locator")
+        ET.SubElement(Locator, "Frequency", Value=frequency)
+        tree = ET.ElementTree(JXY_Export_Set)
+
+        if self.file_name is None:
```

Comment:
self.file_name should be self.project_file_path. You don't need the scattering matrix file name

Reply:
I dont understand. The sonnet file has a name and a path. I need the name

Reply:
self.project_file_path is the full path including the file name. To get the file name follow this snippet:
```
from pathlib import Path
p = Path("/my/path/to/file/file_name.txt")
print(p.stem) # returns file_name
print(p.name)  # returns file_name.txt
```

But you probably want the full path and file name anyways since you're just going to write an xml file to the same directory. Consider 
```
xml_file_name = pathlib.Path(self.project_file_name).with_suffix(".xml")
```